File: examples/Fortran/openmp_converted/Data_Management_wrong_example_omp.f90
program wrong_usage
    use iso_fortran_env, only : INT32, REAL64
    use openacc
use omp_lib
    implicit none
        
    integer(kind=INT32 ), parameter              :: system_size = 10000
    real   (kind=real64), dimension(system_size) :: a, b, c
    integer(kind=INT32 )                         :: i

!  Structured data region
    !$acc data create(a, b) copyout(c)
!$omp target data map(from:c) map(alloc:a,b)

        !$acc parallel loop present(a(:), b(:))
!$omp target teams loop
        do i = 1, system_size
            a(i) = i
            b(i) = i*2
        enddo

        ! We update an element of the array on the CPU
        !$acc update host (a(:))
!$omp target update from(a(:))
        a(12) = 42
        !$acc update device (a(:))
!$omp target update to(a(:))

        !$acc parallel loop present(b(:), c(:)) copyin(a(:))
!$omp target teams loop 
        do i = 1, system_size
            c(i) = a(i) + b(i)
        enddo
    !$acc end data
!$omp end target data
    write(0,"(a22,f10.5)") "value at position 12: ", c(12)
end program wrong_usage

! Code was translated using: intel-application-migration-tool-for-openacc-to-openmp Data_Management_wrong_example.f90

File: examples/C/openmp_converted/Data_Management_wrong_example_omp.c
#include <stdio.h>
#include <stdlib.h>
int main(void){
    int size = 10000;
    int a[size], b[size], c[size];

// Structured data region
    #pragma acc data create(a, b) copyout(c)
#pragma omp target data map(from:c) map(alloc:a,b)
    {
        #pragma acc parallel loop present(a, b)
#pragma omp target teams loop map(present,alloc:a,b)
        for(int i=0; i<size; ++i){
            a[i] = i;
            b[i] = 2*i;
        }
        // We update an element of the array on the CPU 
        a[14] = 78324;

        #pragma acc parallel loop present(b, c) copyin(a)
#pragma omp target teams loop map(to:a) map(present,alloc:b,c)
        for (int i=0; i<size; ++i){
            c[i] = a[i]+b[i];
        }
    }
    printf("value at position 14: %d\n", c[14]); 
}


// Code was translated using: /home/very/bin/acc2mp -async=nowait -specify-language=C -generate-report Data_Management_wrong_example.c
